src/lib/statistics/chunk_statistics/histograms/abstract_histogram.cpp

```diff
@@ -0,0 +1,625 @@
+#include "abstract_histogram.hpp"
+
+#include <cmath>
+
+#include <algorithm>
+#include <limits>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+#include "expression/evaluation/like_matcher.hpp"
+#include "histogram_utils.hpp"
+#include "storage/create_iterable_from_segment.hpp"
+
+#include "constant_mappings.hpp"
+#include "resolve_type.hpp"
+
+namespace opossum {
+
+template <typename T>
+AbstractHistogram<T>::AbstractHistogram() : _supported_characters(""), _string_prefix_length(0ul) {}
+
+template <>
+AbstractHistogram<std::string>::AbstractHistogram() {
+  const auto pair = get_default_or_check_prefix_settings();
+  _supported_characters = pair.first;
+  _string_prefix_length = pair.second;
+}
+
+template <>
+AbstractHistogram<std::string>::AbstractHistogram(const std::string& supported_characters,
+                                                  const uint64_t string_prefix_length)
+    : _supported_characters(supported_characters), _string_prefix_length(string_prefix_length) {
+  DebugAssert(check_prefix_settings(_supported_characters, _string_prefix_length), "Invalid prefix settings.");
+}
+
+template <typename T>
+std::string AbstractHistogram<T>::description() const {
+  std::stringstream stream;
+  stream << histogram_type_to_string.at(histogram_type()) << std::endl;
+  stream << "  distinct    " << total_count_distinct() << std::endl;
+  stream << "  min         " << min() << std::endl;
+  stream << "  max         " << max() << std::endl;
+  // TODO(tim): consider non-null ratio in histograms
+  // stream << "  non-null " << non_null_value_ratio() << std::endl;
+  stream << "  bins        " << num_bins() << std::endl;
+
+  stream << "  edges / counts " << std::endl;
+  for (BinID bin = 0u; bin < num_bins(); bin++) {
+    stream << "              [" << _bin_min(bin) << ", " << _bin_max(bin) << "]: ";
```

Comment:
I think `-` or `->` would look better than the comma here.

Reply:
I am not too opinionated about this, but `[a, b]` is the mathematical notion of an interval, which is exactly what this is. So I'd leave it as is, if it doesn't bother you too much.